### PR TITLE
User being fetched twice during authentication sequence

### LINKED ISSUE #560: User being fetched twice during authentication sequence
# Problem

The user is fetched from the database first during the call to [`plug Guardian.Plug.LoadResource`](https://github.com/code-corps/code-corps-api/blob/develop/web/router.ex#L26) and then it is fetched again during the call to [`plug CodeCorps.Plug.CurrentUser`](https://github.com/code-corps/code-corps-api/blob/develop/web/router.ex#L34).

It may be possible to alleviate the second call by calling `Guardian.Plug.current_resource(conn)` [here](https://github.com/code-corps/code-corps-api/blob/develop/web/plugs/current_user.ex#L12) in order to fetch the user from the conn instead of from the database.

This will probably need some testing to ensure that `Guardian.Plug.current_resource/1` works as expected.

### PR DESCRIPTION
**Pull request by YuriiZdyrko**
# What's in this PR?

- changed plugs/current_user.exs to fetch User from conn
- added unit test

## References
Fixes #560

Progress on: #

### CODE REVIEW

File: web/plugs/current_user.ex [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 
+  @spec call(Plug.Conn.t, Keyword.t) :: Plug.Conn.t
   def call(conn, _opts) do
-    case Guardian.Plug.current_token(conn) do
+    case Guardian.Plug.current_resource(conn) do
+      user = %CodeCorps.User{} ->
+        Plug.Conn.assign(conn, :current_user, user)
       nil ->
         conn
-      current_token ->
```
Comment (begedin) [inline web/plugs/current_user.ex]: I see you removed this completely. Is it your experience that this case does not happen at all? Makes me wonder why we had it in the first place. @JoshSmith Any ideas?

**Review by begedin — COMMENTED** [review]
Looks really good, IMO and definitely good to go, but would like to hear @JoshSmith thoughts on this, since he was the one who originally implemented it.

Any clue why we initially went with `current_token` and not `current_resource`?

**Comment by joshsmith** [comment]
I'm not exactly sure why this is working vs what I was doing manually. Is the user decoded from the token this way? I need to test against the frontend too I think.

**Comment by YuriiZdyrko** [comment]
Looks like previous implementation duplicated part of LoadResource's work.
https://hexdocs.pm/guardian/Guardian.Plug.LoadResource.html

**Comment by joshsmith** [comment]
Interesting! Can you rebase onto develop and I'll get this merged asap?

**Review by joshsmith — APPROVED** [review]

**Comment by YuriiZdyrko** [comment]
Rebased

**Comment by joshsmith** [comment]
@banzay actually the "merge into" button doesn't do a rebase. You'll need to do something in your local branch like `get fetch --all` and `git rebase origin/develop` in order to rebase onto `develop`. Happy to pair on it if you haven't done it, but I think there are some docs in this repo about it.

**Comment by joshsmith** [comment]
@banzay actually figured out that I can do this myself. Learning!

**Comment by YuriiZdyrko** [comment]
👍

**Pull request merged**